release.sh

```diff
@@ -0,0 +1,464 @@
+    read -n 1 user_response
+    printf "\n"
+  done
+  printf $reset
+}
+
+function prompt {
+  user_response=''
+  printf $blue
+  printf "%b" "$*"
+  printf $red
+  read user_response
+  printf $reset
+}
+
+function prompt_password {
+  user_response=''
+  printf $blue
+  printf "%b" "$*"
+  printf $red
+  read -s user_response
+  printf $reset
+}
+
+#----------------------
+# Commands
+#----------------------
+function bump_version {
+
+  message "Stashing current work..."
+
+  stash_ref=$(git stash create)
+  run_message git stash create
+
+  if [[ $stash_ref ]]; then
+    run git reset --hard
+    message "Stashed current work to: $stash_ref"
+  else
+    message "Nothing to stash"
+  fi
+
+  branch_name="$(git symbolic-ref HEAD 2>/dev/null)"
+  branch_name=${branch_name##refs/heads/}
+  message "Current branch: $branch_name"
+
+  if [[ $branch_name != 'master' ]]; then
+    message "Switching to master..."
+    run git checkout master
+  fi
+
+  message "Pulling latest version from GitHub"
+  run git pull --rebase
+
+  confirm "Replace stable tag on readme.txt with $version?"
+  message "Replacing stable tag on readme.txt"
+  run sed -i -e "s/Stable tag: .*/Stable tag: $version/" ./readme.txt
+  run git diff
+  confirm "Add changes to commit?"
+  run git add readme.txt
+  run rm readme.txt-e
+
+  confirm "Replace version on facebook-instant-articles-wp.php with $version?"
+  message "Replacing version on facebook-instant-articles-wp.php"
+  run sed -i -e "s/^ \* Version: .*/ * Version: $version/" facebook-instant-articles.php
+  run sed -i -e "s/define( 'IA_PLUGIN_VERSION', '[0-9.]*' );/define( 'IA_PLUGIN_VERSION', '$version' );/" facebook-instant-articles.php
+  run git diff
+  confirm "Add changes to commit?"
+  run git add facebook-instant-articlesg.php
+  run rm facebook-instant-articles.php-e
+
+  confirm "Commit version bump on master with message 'Bump version to $version'?"
+  run git commit -m "Bump version to $version"
+
+  confirm "Create tag $version?"
+  run git tag $version
+
+  confirm "Push tag and commit to GitHub?"
+  run git push
+  run git push --tags
+
+  revert_repo
+
+  echo "🍺  Tag $version created!"
+}
+
+function release {
+
+  confirm "Create a new release for $version?"
+
+  message "Stashing current work..."
+
+  stash_ref=$(git stash create)
+  run_message git stash create
+
+  if [[ $stash_ref ]]; then
+    run git reset --hard
+    message "Stashed current work to: $stash_ref"
+  else
+    message "Nothing to stash"
+  fi
+
+  branch_name="$(git symbolic-ref HEAD 2>/dev/null)"
+  branch_name=${branch_name##refs/heads/}
+  message "Current branch: $branch_name"
+
+  if [[ $branch_name != 'master' ]]; then
+    message "Switching to master..."
+    run git checkout master
+  fi
+
+  message "Pulling latest version from GitHub"
+  run git pull --rebase
+
+  if [[ ! -e resty ]]; then
+    message "Downloading resty to connect to GitHub API"
+    run curl -L http://github.com/micha/resty/raw/master/resty > resty
```

Comment:
For this and jsawk we should consider pulling from a fixed tag/releases, hopefully this will avoid scenarios where someone is burned by broken master branches